### PR TITLE
Dockcross and android-arm

### PR DESCRIPTION
**Pull request by brainstorm**
Changes needed to dockross-compile android-arm with radare2-release.

Please make sure to review the following:

https://github.com/radare/radare2/pull/6319#issuecomment-276046516

### CODE REVIEW

File: mk/android.mk [hunk 1]
```diff
@@ -1,7 +1,8 @@
-#CC=arm-linux-androideabi-gcc
-CC=ndk-gcc -fPIC -fPIE
+CC=arm-linux-androideabi-gcc
```
Comment (radare) [inline mk/android.mk]: this is ARM specific, will break the non-arm builds

File: mk/android.mk [hunk 2]
```diff
@@ -1,7 +1,8 @@
-#CC=arm-linux-androideabi-gcc
-CC=ndk-gcc -fPIC -fPIE
```
Comment (radare) [inline mk/android.mk]: dont remove this, add an if to check if the arm-linux--.... is in PATH

File: mk/android.mk [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
+CC=arm-linux-androideabi-gcc
+#CC=ndk-gcc -fPIC -fPIE
 #RANLIB=ndk-ranlib
-USERCC=ndk-gcc -fPIC -fPIE
+USERCC=arm-linux-androideabi-gcc -fPIC -fPIE
```
Comment (radare) [inline mk/android.mk]: same here

File: mk/android.mk (hunk 1 above)
Comment (radare) [inline mk/android.mk]: use $ARCH

File: mk/android.mk [hunk 4]
```diff
@@ -1,7 +1,13 @@
-#CC=arm-linux-androideabi-gcc
+ifeq (${PATH},"${ARCH}-linux-androideabi-gcc")
```
Comment (radare) [inline mk/android.mk]: i dont think PATH will be $arch-...

File: sys/android-shell.sh [hunk 5]
```diff
@@ -45,73 +45,76 @@ ROOT=`dirname $PWD/$0`
...
-export CC
-export PS1
-export AR
-export RANLIB
-A=$@
-if [ -n "$A" ]; then
-	${SHELL} -c "$A"
-else
-	${SHELL}
+    #PATH=$SDK/tools:$SDK/platform-tools:$NDK:${NDKPATH_X86}:${NDKPATH_ARM}:${NDKPATH_MIPS64}:${NDKPATH_AARCH64}:${NDKPATH_MIPS}:$PATH
```
Comment (radare) [inline sys/android-shell.sh]: use tabs

File: mk/android.mk [hunk 6]
```diff
@@ -1,7 +1,13 @@
-#CC=arm-linux-androideabi-gcc
+ifeq (${PATH},"/usr/arm-linux-androideabi/bin/${ARCH}-linux-androideabi-gcc")
```
Comment (radare) [inline mk/android.mk]: wat is this condition even happening at all? it looks still wrong to me

File: sys/android-shell.sh [hunk 7]
```diff
@@ -45,21 +45,24 @@ ROOT=`dirname $PWD/$0`
...
+	echo "Building android locally with NDK instead of dockcross..."
+	# TODO: autodetect or gtfo
+	if [ -f ~/.r2androidrc ]; then
+		. ~/.r2androidrc
+		echo "Using data from ${HOME}/.r2androidrc.."
+	else
+		#[ -z "${SDK}" ] && SDK=${HOME}/Downloads/android-sdk-${OS}
+		if [ -z "${NDK}" ]; then
+			if [ "`uname`" = "Darwin" ]; then
+				NDK=${HOME}/Library/Android/sdk/ndk-bundle/
```
Comment (radare) [inline sys/android-shell.sh]: quote, if HOME have spaces something funky may happen

File: sys/android-shell.sh [hunk 8]
```diff
@@ -45,21 +45,24 @@ ROOT=`dirname $PWD/$0`
...
+			else
+				NDK=${HOME}/Downloads/android-ndk-r7b
+			fi
 		fi
+		[ -z "${NDK}" ] && NDK=${HOME}/Downloads/android-ndk-r7b
```
Comment (radare) [inline sys/android-shell.sh]: quote

File: sys/android-shell.sh [hunk 9]
```diff
@@ -76,44 +79,45 @@ if [ "${BUILD}" != 0 ]; then
...
+  chmod +x ${NDK}/ndk-gcc
+  CC=ndk-gcc
+  PS1="[r2-android-${NDK_ARCH}]> "
+  export CC
+  export PS1
+  export AR
+  export RANLIB
+  A=$@
+  if [ -n "$A" ]; then
+    ${SHELL} -c "$A"
```
Comment (radare) [inline sys/android-shell.sh]: use tabs

File: mk/android.mk (hunk 6 above)
Comment (brainstorm) [inline mk/android.mk]: That is the path to gcc as it is inside dockcross :-S How would you specify this otherwise?

**Comment by radare** [comment]
conflict. can you rebase?

File: mk/android.mk (hunk 6 above)
Comment (radare) [inline mk/android.mk]: put a comment in this conditional to make that clear

File: mk/android.mk [hunk 10]
```diff
@@ -47,6 +53,7 @@ ONELIB=0
 OSTYPE=android
 LINK=
 #CC_AR=ndk-ar -r ${LIBAR}
+CC_AR=arm-linux-androideaby-ar -r ${LIBAR}
```
Comment (radare) [inline mk/android.mk]: uh. that should be only for dockcross? it shuoldbe fine for the sdk i guess

also.. i doubt androideabY exists. it's 'i'

File: mk/android.mk [hunk 11]
```diff
@@ -1,7 +1,13 @@
-#CC=arm-linux-androideabi-gcc
+ifeq (${PATH},"/usr/arm-linux-androideabi/bin/${ARCH}-linux-androideabi-gcc")
+CC=${ARCH}-linux-androideabi-gcc
+RANLIB={ARCH}-linux-androideabi-ranlib
+USERCC=${ARCH}-linux-androideabi-gcc -fPIC -fPIE
+else
 CC=ndk-gcc -fPIC -fPIE
-#RANLIB=ndk-ranlib
+RANLIB=ndk-ranlib
```
Comment (radare) [inline mk/android.mk]: there's no ndk-ranlib. so dont uncomment this

**Pull request merged**